File: tests/fixtures/cover-behaviour-gate/mock_only.py
# Fixture for tests/unit/test-cover-behaviour-gate.mjs: rule (a), mock-call assertions only.
from unittest.mock import MagicMock, patch

from app.checkout import checkout


def test_calls_gateway():
    gateway = MagicMock()
    checkout(gateway, total=10)
    gateway.charge.assert_called_once_with(10)


def test_logs_once():
    log = MagicMock()
    checkout(MagicMock(), total=0, log=log)
    assert log.call_count == 1


@patch("app.checkout.audit")
def test_audits(audit):
    checkout(MagicMock(), total=1)
    assert audit.called
    audit.assert_has_calls([])

File: tests/fixtures/cover-behaviour-gate/mock-only.ts
// Fixture for tests/unit/test-cover-behaviour-gate.mjs: rule (a), mock-call assertions only.
import { expect, it, vi } from 'vitest';
import { checkout } from '../src/checkout';

it('calls the payment gateway', async () => {
  const gateway = { charge: vi.fn() };
  await checkout(gateway, { total: 10 });
  expect(gateway.charge).toHaveBeenCalledWith(10);
});

it('logs once', () => {
  const log = vi.fn();
  checkout({ charge: vi.fn() }, { total: 0 }, log);
  expect(log).toHaveBeenCalledTimes(1);
  expect(log).not.toHaveBeenCalledWith('error');
});

it('passes the total through', () => {
  const charge = vi.fn();
  checkout({ charge }, { total: 5 });
  expect(charge.mock.calls[0][0]).toBe(5);
});
